recipes/fixed_math/all/conanfile.py

```diff
@@ -0,0 +1,142 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.files import copy, get, rmdir, replace_in_file
+from conan.tools.scm import Version
+from conan.tools.layout import basic_layout
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.microsoft import is_msvc
+import os
+
+required_conan_version = ">=1.53.0"
+
+class FixedMathConan(ConanFile):
+    name = "fixed_math"
+    description = "A High-Performance C++17 Library for Fixed-Point 48.16 Arithmetic"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/arturbac/fixed_math/"
+    topics = ("mathematics", "fixed-point")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "header_only": [True, False],
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "header_only": False,
+        "shared": False,
+        "fPIC": True,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "10", # fixed_math requires __has_builtin
+            "clang": "7",
+            "apple-clang": "12",
+            "Visual Studio": "16",
+            "msvc": "192",
+        }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.header_only:
+            self.options.rm_safe("shared")
+            self.options.rm_safe("fPIC")
+            self.package_type = "header-library"
+        elif self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        if self.options.header_only:
+            basic_layout(self, src_folder="src")
+        else:
+            cmake_layout(self, src_folder="src")
+
+    def package_id(self):
+        if self.info.options.header_only:
+            self.info.clear()
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+        if self.settings.os == "Windows" and self.options.get_safe("shared"):
+            raise ConanInvalidConfiguration(f"{self.ref} does not support shared builds on Windows")
```

Comment:
As suggested by @uilianries in another PR, this can be replaced with
```py
    self.package_type = "static-library"
    del self.options.shared
```
under `config_options()`, which allows the package to be used with `*/*:shared=True` on Windows.

Reply:
@valgur 
Thank a lot!
Now I see. Fixed.